ygen/ir.go

```diff
@@ -89,6 +89,45 @@ type LangMapper interface {
 	// the mapped such that leaves of type leafref can be resolved to
 	// their target leaves.
 	SetSchemaTree(*schemaTree)
+
+	// LangMapperExt contains extensions that the LangMapper instance
+	// should implement if extra information from the IR is required.
+	// When implementing this, UnimplementedLangMapperExt should be
+	// embedded in the implementation type in order to ensure forward
+	// compatibility.
+	LangMapperExt
+}
+
+// LangMapperExt contains extensions that the LangMapper instance should
+// implement if extra information from the IR is required. These flag values
+// are expected to contain information rarely used but needed from goyang's
+// AST. Values that are expected to be used more often should be placed in the
+// IR itself so that other users can get access to the same information without
+// implementing it themselves.
+type LangMapperExt interface {
+	// PopulateFieldFlags populates extra information given a particular
+	// field of a ParsedDirectory and the corresponding AST node.
```

Comment:
Is this function called for all fields whether they are a leaf, leaf-list, list or container? What about for choice/case? It'd be good to just ensure that we have a clear expectation here.

Reply:
Fixed comment.

Reply:
Thanks.